Remotion/Data/DomainObjects/Mapping/ClassDefinition.cs

```diff
@@ -120,66 +90,52 @@ public bool IsSameOrBaseClassOf (ClassDefinition classDefinition)
       return false;
     }
 
+    public override TypeDefinition[] GetTypeHierarchy ()
+    {
+      var allDerivedClasses = new List<ClassDefinition>();
+      allDerivedClasses.Add(this);
+      FillAllDerivedClasses(allDerivedClasses);
+
+      var allDerivedTypes = new TypeDefinition[allDerivedClasses.Count];
+      for (var i = 0; i < allDerivedTypes.Length; i++)
+        allDerivedTypes[i] = allDerivedClasses[i];
+
+      return allDerivedTypes;
+    }
+
     public ClassDefinition[] GetAllDerivedClasses ()
     {
       var allDerivedClasses = new List<ClassDefinition>();
       FillAllDerivedClasses(allDerivedClasses);
       return allDerivedClasses.ToArray();
     }
 
-    public ClassDefinition GetInheritanceRootClass ()
+    public override TypeDefinition GetInheritanceRoot ()
     {
       if (BaseClass != null)
-        return BaseClass.GetInheritanceRootClass();
+        return BaseClass.GetInheritanceRoot();
 
       return this;
     }
 
-    [Obsolete("Contains (...) method was ambiguous between GetPropertyDefinitions() and MyPropertyDefinitions. Use GetPropertyDefinitions().Contains (...) instead. (Version: 3.0.0)", true)]
+    [Obsolete(
+        "Contains (...) method was ambiguous between GetPropertyDefinitions() and MyPropertyDefinitions. Use GetPropertyDefinitions().Contains (...) instead. (Version: 3.0.0)",
+        true)]
     public bool Contains (PropertyDefinition propertyDefinition)
     {
       throw new NotSupportedException("Use GetPropertyDefinitions().Contains (...) instead. (Version: 3.0.0)");
     }
 
-    public PropertyDefinitionCollection GetPropertyDefinitions ()
-    {
-      return _cachedPropertyDefinitions.Value;
-    }
-
-    public RelationEndPointDefinitionCollection GetRelationEndPointDefinitions ()
-    {
-      return _cachedRelationEndPointDefinitions.Value;
-    }
-
-    public IRelationEndPointDefinition? GetRelationEndPointDefinition (string propertyName)
-    {
-      ArgumentUtility.CheckNotNullOrEmpty("propertyName", propertyName);
-
-      return _cachedRelationEndPointDefinitions.Value[propertyName];
-    }
-
-    public IRelationEndPointDefinition GetMandatoryRelationEndPointDefinition (string propertyName)
-    {
-      ArgumentUtility.CheckNotNullOrEmpty("propertyName", propertyName);
-
-      IRelationEndPointDefinition? relationEndPointDefinition = GetRelationEndPointDefinition(propertyName);
-
-      if (relationEndPointDefinition == null)
-        throw CreateMappingException("No relation found for class '{0}' and property '{1}'.", ID, propertyName);
-
-      return relationEndPointDefinition;
-    }
-
     public bool IsMyRelationEndPoint (IRelationEndPointDefinition relationEndPointDefinition)
     {
-      ArgumentUtility.CheckNotNull("relationEndPointDefinition", relationEndPointDefinition);
+      ArgumentUtility.CheckNotNull(nameof(relationEndPointDefinition), relationEndPointDefinition);
```

Comment:
I would prefer not to do this sort of cleanup in between. it's better to do one global cleanup for this.